lib/appsignal.rb

```diff
@@ -235,22 +235,26 @@ def logger
     end
 
     def log_formatter
-        proc do |severity, datetime, progname, msg|
-          "[#{datetime.strftime('%Y-%m-%dT%H:%M:%S')} (process) ##{Process.pid}][#{severity}] #{msg}\n"
-        end
+      proc do |severity, datetime, progname, msg|
+        "[#{datetime.strftime('%Y-%m-%dT%H:%M:%S')} (process) ##{Process.pid}][#{severity}] #{msg}\n"
+      end
     end
 
     def start_logger(path_arg=nil)
       path = Appsignal.config ? Appsignal.config.log_file_path : nil
-      if path && !ENV['DYNO']
-        @logger = Logger.new(path)
-        @logger.formatter = log_formatter
-      else
-        @logger = Logger.new($stdout)
-        @logger.formatter = lambda do |severity, datetime, progname, msg|
-          "appsignal: #{msg}\n"
+      if path && !heroku?
+        begin
+          @logger = Logger.new(path)
+          @logger.formatter = log_formatter
+        rescue Exception => error
```

Comment:
Although I agree that we should be careful here, we probably [don't want to](http://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby) `rescue Exception` and turn it into a warning.